ste/sourceInfoProvider-Local_linux.go

```diff
@@ -0,0 +1,163 @@
+// +build linux
+
+package ste
+
+import (
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	"golang.org/x/sys/unix"
+	"time"
+)
+
+func (f localFileSourceInfoProvider) HasUNIXProperties() bool {
+	return true
+}
+
+func (f localFileSourceInfoProvider) GetUNIXProperties() (common.UnixStatAdapter, error) {
+	{ // attempt to call statx, if ENOSYS is returned, statx is unavailable
+		var stat unix.Statx_t
+		// dirfd is a null pointer, because we should only ever be passing relative paths here, and directories will be passed via transferInfo.Source.
+		// AT_SYMLINK_NOFOLLOW is not used, because we automagically resolve symlinks. TODO: Add option to not follow symlinks, and use AT_SYMLINK_NOFOLLOW when resolving is disabled.
+		err := unix.Statx(0, f.transferInfo.Source,
+			unix.AT_STATX_SYNC_AS_STAT, // We want to sync attributes to ensure correctness.
+			unix.STATX_BTIME,           // Let's pull only the special statx properties, and yank the rest from a standard stat_t call.
```

Comment:
this should be STATX_ALL, now that we are using only statx().
also the " // We want to sync attributes to ensure correctness." comment is not correct, as AT_STATX_SYNC_AS_STAT will usually not sync.